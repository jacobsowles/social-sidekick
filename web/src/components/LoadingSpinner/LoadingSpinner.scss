@import '@styles/_variables.scss';

.loading-spinner {
  color: $brand-primary;
}
