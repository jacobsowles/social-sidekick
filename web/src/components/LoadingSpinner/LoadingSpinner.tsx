import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

import './LoadingSpinner.scss';

export interface LoadingSpinnerProps {
  className?: string;
  size?:
    | 'xs'
    | 'sm'
    | 'lg'
    | '1x'
    | '2x'
    | '3x'
    | '4x'
    | '5x'
    | '6x'
    | '7x'
    | '8x'
    | '9x'
    | '10x'
    | undefined;
}

const LoadingSpinner: FunctionComponent<LoadingSpinnerProps> = ({ className, size, ...rest }) => {
  return (
    <FontAwesomeIcon
      className={classNames('loading-spinner', className)}
      icon={['fas', 'spinner']}
      size={size}
      {...rest}
      spin
    />
  );
};

LoadingSpinner.defaultProps = {
  className: undefined,
  size: '4x'
};

export default LoadingSpinner;
