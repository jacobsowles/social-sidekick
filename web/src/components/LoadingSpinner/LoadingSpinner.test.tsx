import React from 'react';

import { init, shallow, toJson } from '@tests/component-test-base';
import LoadingSpinner from './LoadingSpinner';

describe('LoadingSpinner', () => {
  init(() => <LoadingSpinner />);

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });
});
