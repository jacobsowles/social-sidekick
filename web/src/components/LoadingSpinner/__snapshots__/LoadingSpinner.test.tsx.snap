// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner should render the component without crashing 1`] = `
<FontAwesomeIcon
  border={false}
  className="loading-spinner"
  fixedWidth={false}
  flip={null}
  icon={
    Array [
      "fas",
      "spinner",
    ]
  }
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size="4x"
  spin={true}
  symbol={false}
  title=""
  transform={null}
/>
`;
